File: backend/app/schemas/order.py
from pydantic import BaseModel, EmailStr
from datetime import datetime
from typing import Optional


class OrderBase(BaseModel):
    first_name: str
    last_name: str
    address: Optional[str]
    phone_number: str
    billing_addess: str
    region: str
    country: str
    city: str
    document: Optional[str]
    email: str
    extra_info: Optional[str]
    de: Optional[str]
    para: Optional[str]
    isGift: bool
    state: str
    quantity: int
    shipping_cost: float
    total: float
    subtotal: float
    shipping_guide: Optional[str] = "No asignada"
    shipping_guide_url: Optional[str] = ""
    shipping_guide_number: Optional[str] = ""
    paid_status: Optional[str] = "Pendiente"
    preference: str
    pyment_id: Optional[str] = ""



class OrderCreate(OrderBase):
    pass


class OrderUpdate(OrderBase):
    pass


class OrderInDBBase(OrderBase):
    id: int

    class Config:
        orm_mode: True


class Order(OrderInDBBase):
    pass


class OrderInDB(OrderInDBBase):
    pass

File: itsocks-laravel/app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Order extends Model
{
    use HasFactory;
    // Nota: tabla 'orders' (plural — es la tabla de Laravel, no la singular de FastAPI)
    // FastAPI usaba '__tablename__ = order' (singular)
    protected $table = 'orders';

    protected $fillable = [
        'customer_name',
        'first_name',
        'last_name',
        'email',
        'phone',
        'document',
        'document_type',
        'shipping_city',
        'shipping_department',
        'shipping_address',
        'billing_address',   // CORREGIDO: era billing_addess en FastAPI
        'payment_id',        // CORREGIDO: era pyment_id en FastAPI
        'preference_id',
        'paid_status',
        'gift_from',         // CORREGIDO: era 'de' en FastAPI
        'gift_to',           // CORREGIDO: era 'para' en FastAPI
        'is_gift',
        'gift_message',
        'shipping_guide',
        'shipping_guide_url',
        'shipping_guide_number',
        'tracking_number',
        'subtotal',
        'shipping_cost',
        'discount_amount',
        'total',
        'status',
        'country',
        'extra_info',
        'special_instructions',
        'quantity',
    ];

    protected $casts = [
        'subtotal' => 'decimal:2',
        'shipping_cost' => 'decimal:2',
        'discount_amount' => 'decimal:2',
        'total' => 'decimal:2',
        'is_gift' => 'boolean',
        'quantity' => 'integer',
    ];

    public function scopePending($query)
    {
        return $query->where('status', 'pending');
    }

    public function scopePaid($query)
    {
        return $query->where('status', 'paid');
    }

    public function scopeShipped($query)
    {
        return $query->where('status', 'shipped');
    }

    public function productOrders(): HasMany
    {
        return $this->hasMany(ProductOrder::class, 'order_id');
    }
}
